### PR TITLE
bootloader: make screen messages more user friendly

### PR DESCRIPTION
**Pull request by x1ddos**
Some consider "bootloader" text on the screen confusing, so the
bootloader now invites users to follow instructions on the
BitBoxApp screen, consistent with the regular mode once a firmware is
booted.

In the case where app flash is empty, it also displays a "let's get
started" message, as well as rendering chunk writes progress with
"INSTALLING" label. BitBox02 is shipped with empty app flash. It can
also be erased manually via the bootloader protocol.

### CODE REVIEW

**Comment by x1ddos** [comment]
ok, I reverted changes in `_api_firmware_erase`. @benma convinced me it was mostly useless.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## [Unreleased]
 - Display granular error codes when unlock fails unexpectedly
+- Make bootloader mode more user-friendly
```
Comment (benma) [inline CHANGELOG.md:5]: Bootloader releases are separate from firmware releases (different versions and cycle) - maybe it's best to add a separate changelog or separate section for the bootloader in this changelog?

File: src/bootloader/bootloader.c [hunk 2]
```diff
@@ -160,6 +160,26 @@ typedef union {
 
 static uint8_t _loading_ready = 0;
 static uint8_t _firmware_num_chunks = 0;
+// Indicates whether the whole app flash contains only 0xFF.
+// This controls bootloader text messages on the screen.
+// The value is computed at bootloader enter and at OP_ERASE behind a condition.
+static uint8_t _is_app_flash_empty = 0;
```
Comment (benma) [inline src/bootloader/bootloader.c:166]: we usually use `bool` for bools :P

**Review by benma — COMMENTED** [review]
nice! utACK, but I want to test it first

File: src/bootloader/bootloader.c (hunk 2 above)
Comment (x1ddos) [reply src/bootloader/bootloader.c:166]: that's not what I'm seeing in the code:

- `_loading_read` is treated as bool
- some vars use `sectrue_u8` and `secfalse_u8`
- others do use false

wanted to be consistent with the other static vars.

**Comment by benma** [comment]
Please bump the bootloader patch version as well.

**Comment by x1ddos** [comment]
> Please bump the bootloader patch version as well.

done

**Comment by x1ddos** [comment]
PTAL

File: src/bootloader/bootloader.c (hunk 2 above)
Comment (benma) [reply src/bootloader/bootloader.c:166]: Great you found the only other example :D ok fine - but consider changing both to `bool` :)

**Review by benma — APPROVED** [review]
tACK.

from a UI perspective I have a feeling putting the text and logo into the corners is not good design. Centering them and having some padding would look more normal to me. 

Not in this PR of course, but maybe sth for @jadzeidan to consider.

**Comment by jadzeidan** [comment]
> tACK.
> 
> from a UI perspective I have a feeling putting the text and logo into the corners is not good design. Centering them and having some padding would look more normal to me.
> 
> Not in this PR of course, but maybe sth for @jadzeidan to consider.

The corners seem fine to me, but i'll have a better look once I load the new bootloader onto my device.

**Comment by x1ddos** [comment]
Oh, forgot to attach a photo. Here's how it looks like on empty firmware. I like it. 
image

File: src/bootloader/bootloader.c (hunk 2 above)
Comment (x1ddos) [reply src/bootloader/bootloader.c:166]: :) ok, will send you a separate PR

**Comment by thisconnect** [comment]
Looks great 👍

File: src/bootloader/bootloader.c (hunk 2 above)
Comment (x1ddos) [reply src/bootloader/bootloader.c:166]: https://github.com/digitalbitbox/bitbox02-firmware/pull/760

**Pull request merged**